File: src/ProjectR.Scripting/content/scripts/spells/FieryFuryFist.lua
function GetTargetType()
    return Single
end

function GetName()
  return "Fiery Fury Fist"
end

function GetDescription()
    return "Strike with your flame-imbued fist.\nThe heat causes you to ignore some of the enemies DEF."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .58
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  local damage = (5 * aAD * (aFIR/100) - .8 * dDEF) * (100/dFIR)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/FieryFuryFist.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class FieryFuryFist : SpellScriptBase
    {
        public override string Name { get { return "Fiery Fury Fist"; } }
        public override string Description { get { return "Strike with your flame-imbued fist.\nThe heat causes you to ignore some of the enemies DEF."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 58; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (5 * cMD * (cFIR / 100) - .8 * tMR) * (100 / tFIR);
            DealDamage(damage);
        }
    }
}